enterprise/frontend/src/metabase-enterprise/audit_app/components/AuditNotificationEditModal/AuditNotificationEditModal.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { t } from "ttag";
+import Button from "metabase/components/Button";
+import FormMessage from "metabase/components/form/FormMessage";
+import ModalContent from "metabase/components/ModalContent";
+import UserPicker from "metabase/components/UserPicker";
+
+const propTypes = {
+  item: PropTypes.object.isRequired,
+  type: PropTypes.oneOf(["alert", "pulse"]).isRequired,
+  users: PropTypes.array.isRequired,
+  onUpdate: PropTypes.func,
+  onDelete: PropTypes.func,
+  onClose: PropTypes.func,
+};
+
+const AuditNotificationEditModal = ({
+  item,
+  type,
+  users,
+  onUpdate,
+  onDelete,
+  onClose,
+}) => {
+  const [channels, setChannels] = useState(item.channels);
+  const [error, setError] = useState();
+  const hasRecipients = channels.some(c => c.recipients.length > 0);
+
+  const handleRecipientsChange = (recipients, index) => {
+    const newChannels = [...channels];
+    newChannels[index] = { ...channels[index], recipients };
+    setChannels(newChannels);
+  };
+
+  const handleUpdateClick = async () => {
+    try {
+      await onUpdate(item, channels);
+      onClose(true);
+    } catch (error) {
+      setError(error);
+    }
+  };
+
+  const handleDeleteClick = () => {
+    onDelete(item);
+  };
+
+  const handleClose = () => onClose(true);
+
+  return (
+    <ModalContent
+      title={getTitleMessage(item, type)}
+      footer={[
```

Comment:
Same thought about React.Fragment and extracting the declaration of what goes into the footer prop.